web/packages/teleport/src/types.ts

```diff
@@ -48,9 +50,20 @@ export interface TeleportFeature {
   category?: NavigationCategory;
   section?: ManagementSection;
   hasAccess(flags: FeatureFlags): boolean;
+  // route defines react router Route fields.
+  // This field can be left undefined to indicate
+  // this feature is a parent to children features
+  // eg: FeatureAccessRequests is parent to sub features
+  // FeatureNewAccessRequest and FeatureReviewAccessRequests.
+  // These childrens will be responsible for routing.
   route?: TeleportFeatureRoute;
   navigationItem?: TeleportFeatureNavigationItem;
   topMenuItem?: TeleportFeatureNavigationItem;
+  // isLockedAndUpdatedRouteAndNavigationItem will update this feature's
+  // `route` and `navigationItem` to an alternate route and component if
+  // this feature is locked. Returns true to indicate feature was locked
+  // and changes have been made.
+  isLockedAndUpdatedRouteAndNavigationItem?(ctx: TeleportContext): boolean;
```

Comment:
i would just send only the locked feature flags here (just like `hasAccess` sends only feature flags)